sycamore/utils/extract_json.py

```diff
@@ -1,11 +1,15 @@
 import re
 import json
+from json import JSONDecodeError
 
 
 def extract_json(payload):
-    pattern = r"```json([\s\S]*?)```"
-    match = re.match(pattern, payload)
-    if match:
-        return json.loads(match.group(1))
-    else:
-        raise ValueError("JSON block not found in LLM response")
+    try:
+        return json.loads(payload)
+    except (ValueError, TypeError, JSONDecodeError):
+        pattern = r"```json([\s\S]*?)```"
+        match = re.match(pattern, payload)
+        if match:
+            return json.loads(match.group(1))
+        else:
+            raise ValueError("JSON block not found in LLM response: " + str(payload))
```

Comment:
What's the difference between these (besides the better error message)?

Reply:
we try to first parse the string as json directly, if that fails then we try to parse it as:
```json
{actual json}
```
which is what openai/guidance seems to be returning sometimes